addons/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/operation/SyncAuditLogsFromStorageOperation.java

```diff
@@ -0,0 +1,160 @@
+import org.nuxeo.ecm.platform.audit.listener.StreamAuditEventListener;
+import org.nuxeo.ecm.platform.audit.service.NXAuditEventsService;
+import org.nuxeo.lib.stream.computation.Record;
+import org.nuxeo.lib.stream.computation.Watermark;
+import org.nuxeo.lib.stream.log.LogAppender;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamService;
+
+/**
+ * @author <a href="mailto:kchang@nuxeo.com">Ku Chang</a>
+ *
+ *         Retrieve audit logs from DB backup then populate them to audit
+ *         backend (ES in production)
+ *
+ */
+@Operation(id = SyncAuditLogsFromStorageOperation.ID, category = Constants.W_AUDIT_EVENT, label = "Audit Event: Migrate Audit Storage Events", description = "Migrate directory audit storage logs into audit backend")
+public class SyncAuditLogsFromStorageOperation extends RestoreOperation {
+
+    public static final String ID = "AuditStorage.Sync";
+
+    public static final String RESTORE_AUDIT_STORAGE_LOG_KEY = "restore_audit_storage_log_key";
+    public static final String RESTORE_AUDIT_STORAGE_LOG_VALUE = "restore_audit_storage_log_value";
+
+    @Context
+    protected NXAuditEventsService auditService;
+
+    @Param(name = "after", required = false)
+    protected Long after = Long.MIN_VALUE;
+
+    @Param(name = "batchSize", required = false)
+    protected Integer batchSize = DEAFULT_BATCH_SIZE;
+
+    @Param(name = "keepAlive", required = false)
+    protected Integer keepAlive = DEFAULT_KEEP_ALIVE_SECONDS;
+
+    public static LogEntry getLogEntryFromJson(String jsonEntry) {
+        try {
+            return OBJECT_MAPPER.readValue(jsonEntry, LogEntryImpl.class);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static LogAppender<Record> getLogAppender() {
+        StreamService streamService = Framework.getService(StreamService.class);
+        return streamService.getLogManager(getLogConfig()).getAppender(StreamAuditEventListener.STREAM_NAME);
+    }
+
+    protected static String getLogConfig() {
+        return Framework.getProperty(StreamAuditEventListener.AUDIT_LOG_CONFIG_PROP,
+                StreamAuditEventListener.DEFAULT_LOG_CONFIG);
+    }
+
+    protected static long getTimestampForEntry(LogEntry entry) {
+        if (entry.getEventDate() != null) {
+            return entry.getEventDate().getTime();
+        }
+        return System.currentTimeMillis();
+    }
+
+    protected static String asJson(LogEntry entry) {
+        if (entry == null) {
+            return null;
+        }
+        RenderingContext ctx = RenderingContext.CtxBuilder.get();
+        try {
+            return MarshallerHelper.objectToJson(entry, ctx);
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    protected static void writeEntry(LogAppender<Record> appender, LogEntry entry) {
+        String json = asJson(entry);
+        if (json == null) {
+            return;
+        }
```

Comment:
You shouldn't have null checks in both `asJson` and `writeEntry`. Is it possible at all to have a null `entry` here? If not just check and throw `NuxeoException` but don't put null checks all over the place.